src/engine/Service.cpp

```diff
@@ -200,6 +171,58 @@ ResultTable Service::computeResult() {
   return {std::move(idTable), resultSortedOn(), std::move(localVocab)};
 }
 
+std::optional<std::string> Service::getSiblingValuesClause() const {
+  if (siblingTree_ == nullptr) {
+    return std::nullopt;
+  }
+
+  const auto& siblingResult = siblingTree_->getResult();
+  if (siblingResult->size() >
+      RuntimeParameters().get<"service-max-value-rows">()) {
+    return std::nullopt;
+  }
+
+  std::vector<ColumnIndex> commonColumnIndices;
+  const auto& siblingVars = siblingTree_->getVariableColumns();
+  std::string vars = "";
+  for (const auto& localVar : parsedServiceClause_.visibleVariables_) {
+    auto it = siblingVars.find(localVar);
+    if (it == siblingVars.end()) {
+      continue;
+    }
+    vars += it->first.name() + " ";
+    commonColumnIndices.push_back(it->second.columnIndex_);
+  }
+  vars.pop_back();
+  if (commonColumnIndices.size() > 1) {
+    vars = "(" + vars + ")";
+  }
```

Comment:
How about you ALWAYS add the parentheses, they also work for only one variable, and this makes the code simpler.